packages/protocol/contracts/stability/Broker.sol

```diff
@@ -0,0 +1,229 @@
+    require(checkIsExchangeManager(exchangeManager), "ExchangeManager does not exist");
+    return
+      amountOut = IExchangeManager(exchangeManager).getAmountOut(
+        exchangeId,
+        tokenIn,
+        tokenOut,
+        amountIn
+      );
+  }
+
+  /**
+   * @notice Execute a token swap with fixed amountIn
+   * @param exchangeManager the address of the exchange manager for the pair
+   * @param exchangeId The id of the exchange to use
+   * @param tokenIn The token to be sold
+   * @param tokenOut The token to be bought
+   * @param amountIn The amount of tokenIn to be sold
+   * @param amountOutMin Minimum amountOut to be received - controls slippage
+   * @return amountOut The amount of tokenOut to be bought
+   */
+  function swapIn(
+    address exchangeManager,
+    bytes32 exchangeId,
+    address tokenIn,
+    address tokenOut,
+    uint256 amountIn,
+    uint256 amountOutMin
+  ) external returns (uint256 amountOut) {
+    require(checkIsExchangeManager(exchangeManager), "ExchangeManager does not exist");
+    return
+      amountOut = IExchangeManager(exchangeManager).swapIn(
+        exchangeId,
+        tokenIn,
+        tokenOut,
+        amountIn,
+        amountOutMin
+      );
+    emit Swap(exchangeManager, exchangeId, msg.sender, tokenIn, tokenOut, amountIn, amountOut);
+  }
+
+  /**
+   * @notice Execute a token swap with fixed amountOut
+   * @param exchangeManager the address of the exchange manager for the pair
+   * @param exchangeId The id of the exchange to use
+   * @param tokenIn The token to be sold
+   * @param tokenOut The token to be bought
+   * @param amountOut The amount of tokenOut to be bought
+   * @param amountInMax Maximum amount of tokenIn that can be traded
+   * @return amountIn The amount of tokenIn to be sold
+   */
+  function swapOut(
+    address exchangeManager,
+    bytes32 exchangeId,
+    address tokenIn,
+    address tokenOut,
+    uint256 amountOut,
+    uint256 amountInMax
+  ) external returns (uint256 amountIn) {
+    require(checkIsExchangeManager(exchangeManager), "ExchangeManager does not exist");
+    return
+      amountIn = IExchangeManager(exchangeManager).swapOut(
+        exchangeId,
+        tokenIn,
+        tokenOut,
+        amountOut,
+        amountInMax
+      );
+    emit Swap(exchangeManager, exchangeId, msg.sender, tokenIn, tokenOut, amountIn, amountOut);
+  }
+
+  /* ==================== View Functions ==================== */
+
+  /**
+   * @notice Checks if ExchangeManager exists in the exchangeManagers list
+   * @param exchangeManager the address of the exchange manager for the pair
+   * @return bool Returns true or false
+   */
+  function checkIsExchangeManager(address exchangeManager) public view returns (bool) {
+    for (uint256 i = 0; i < exchangeManagers.length; i++) {
+      if (exchangeManagers[i] == exchangeManager && isExchangeManager[exchangeManagers[i]]) {
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
We can get rid of this function completely, the mapping being public exposes a view function `isExchangeManager(address)` which accomplishes the same thing.